sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -798,7 +804,23 @@ public FileStore getFileStore(Path path) throws IOException {
      */
     @Override
     public void checkAccess(Path path, AccessMode... accessModes) throws IOException {
-        throw new UnsupportedOperationException();
+        if (accessModes != null && accessModes.length != 0) {
+            throw logger.logThrowableAsError(
+                new AccessDeniedException("The access cannot be determined."));
+        }
+        AzurePath.ensureFileSystemOpen(path);
+
+        // Read attributes already wraps BlobStorageException in an IOException.
+        try {
+            readAttributes(path, BasicFileAttributes.class);
+        } catch(IOException e) {
+            if (e.getCause() != null && e.getCause() instanceof BlobStorageException
+                && BlobErrorCode.BLOB_NOT_FOUND.equals(((BlobStorageException) e.getCause()).getErrorCode())) {
```

Comment:
Could this also be ContainerNotFound?

Reply:
Once I add support for containers, yes. A good reminder that I'll have to update a lot of error handling like that